tests/ingress_test.go

```diff
@@ -0,0 +1,170 @@
+package tests
+
+import (
+	"testing"
+
+	netv1 "k8s.io/api/networking/v1"
+	"k8s.io/utils/pointer"
+
+	"github.com/jinzhu/copier"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIngress(t *testing.T) {
+	chart := LoadChart(t)
+
+	tests := []struct {
+		Name       string
+		ValuesFunc func(v *CoderValues)
+		AssertFunc func(t *testing.T, ingress *netv1.Ingress)
+	}{
+		{
+			Name: "simple-ingress",
+			ValuesFunc: func(v *CoderValues) {
+				v.Ingress.Enable = pointer.Bool(true)
+				v.Ingress.Host = pointer.String("install.coder.com")
+				v.Coderd.DevURLsHost = pointer.String("*.install.coder.app")
+			},
+			AssertFunc: func(t *testing.T, ingress *netv1.Ingress) {
+				defaultAnnotations := map[string]string{
+					"nginx.ingress.kubernetes.io/proxy-body-size": "0",
+				}
+				require.Equal(t, defaultAnnotations, ingress.Annotations)
+				pathTypePrefix := netv1.PathTypePrefix
+				expectedRules := []netv1.IngressRule{
+					{
+						Host: "install.coder.com",
+						IngressRuleValue: netv1.IngressRuleValue{
```

Comment:
readability (non-blocking): Could you extract this struct to a variable?

Reply:
I'm confused, this whole struct is already a variable. Can you show me an example of what you mean?

Reply:
This is basically what I was thinking; all the values look to be the same:
```
var tcpCoderdRoot = netv1.IngressRuleValue{
    ...
}

...

expectedRules := []netv1.IngressRule{
    {
        Host: "xxx",
        IngressRuleValue: tcpCoderdRoot,
        ...
    },
...
```

Reply:
Extracted out some of the shared stuff, thanks!